src/BuilderPattern/Builder/IVehicleBuilder.cs

```diff
@@ -0,0 +1,21 @@
+﻿namespace BuilderPattern.Builder
+{
+    using ConcreteBuilder;
+    using Product;
+    using Product.Parts;
+
+    public interface IVehicleBuilder
+    {
+        VehicleBuilder WithBrand(Brand brand);
```

Comment:
these methods should return interfaces

Reply:
Yep, totally agree. When I created the interface I forgot to change these.